File: backend/app/schemas/media.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MediaUploadResponse(BaseModel):
    id: int
    public_id: str
    url: str
    markdown: str
    original_filename: str | None = None
    content_type: str
    size_bytes: int
    created_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/media.ts
import { request } from "./client";

export interface MediaUploadResponse {
  id: number;
  public_id: string;
  url: string;
  markdown: string;
  original_filename: string | null;
  content_type: string;
  size_bytes: number;
  created_at: string | null;
}

export async function uploadMediaImage(file: File): Promise<MediaUploadResponse> {
  const formData = new FormData();
  formData.set("file", file);

  return request<MediaUploadResponse>("/api/media", {
    method: "POST",
    body: formData,
  });
}
